### ISSUE TITLE
What the heck is whatcounts_curl/relay.php !?

### ISSUE THREAD

**Issue by phillipadsmith** [TheTyee/design-article.thetyee.ca#178]
Also wondering why /whatcounts_curl/relay.php is being used vs. the existing subscribe service https://github.com/TheTyee/webhook.thetyee.ca-subscribe ?  

As far as I can tell, this PHP file is not on the live site so the form/relay will only work on preview.thetyee.ca, not on thetyee.ca

These kinds of services should always be set-up as stand-alone domains and not dumped into the web directory that is managed by Bricolage for this reason.  ;-)

**Comment by MrBryan** [comment]
Oh yeah that :)  personally I just think that receiving an immediate notification when a user signs up for the publication rather then On the timer which could be minutes is ideal and making use of all the built in automated email response sending etc. seemed so elegant compared to having all of those other steps in play. It's something I can easily just moved to the live, not anything very complex is going on it just relays the simple post a request to whatcounts.

**Comment by MrBryan** [comment]
Not a strong opinion though happy to move it to another way if you are uncomfortable

**Comment by phillipadsmith** [comment]
> ersonally I just think that receiving an immediate notification when a user signs up for the publication rather then On the timer which could be minutes is ideal and making use of all the built in automated email response sending etc. seemed so elegant compared to having all of those other steps in play. It's something I can easily just moved to the live, not anything very complex is going on it just relays the simple post a request to whatcounts.

I don't understand what you're saying, @MrBryan, the Webhook service mentioned above is immediate -- there is no delay. Can you say more?

**Comment by MrBryan** [comment]
Pardon the strange conjugations above I'm using Siri to dictate into my phone at the moment

**Comment by MrBryan** [comment]
Oh I'll have to look at it again then, I thought it put it in a queue in a database which is then picked up by script that  runs every once in a while

**Comment by phillipadsmith** [comment]
> Pardon the strange conjugations above I'm using Siri to dictate into my phone at the moment

It's been hard to parse what you're saying lately... perhaps this is the issue?

**Comment by phillipadsmith** [comment]
> Oh I'll have to look at it again then, I thought it put it in a queue in a database which is then picked up by script that runs every once in a while

No, it is a simply relay like the one you've built, but one that is managed like the rest of the services on the site.

**Comment by phillipadsmith** [comment]
> Not a strong opinion though happy to move it to another way if you are uncomfortable

My discomfort is mixing services into a directory managed by Bricolage, which makes it hard to know what Bricolage is outputting vs. other "magical" directories that appear in there. 

To avoid memory failures, I've always kept any non-Bricolage directories and files out of the Bricolage-managed directory.

Also, you'd have to remember to move that to the thetyee.ca/www directory, which is just one more thing to remember (or forget) when things go live. Not a good practice, in my experience.

**Comment by MrBryan** [comment]
I targeted the webhooks.thetyee.ca/subscribe/ endpoint and it reports sucessful but no automatic response  & confirmation process is done which I think I had discovered before.

I think that unless you pipe it over to their endpoint none of that stuff is sorted you have to use api for sending a confirmation, then a landing page to convert that confirmation to a successful subscribe with another api call

**Comment by MrBryan** [comment]
i find my user on the list correctly with whatcounts, so the app is working & making it through in that regard. My vote is still to just move the curl_relay to live and avoid doing all that separate to the process that is already worked out until we want to tackle that

**Comment by phillipadsmith** [comment]
> I targeted the webhooks.thetyee.ca/subscribe/ endpoint and it reports sucessful but no automatic response & confirmation process is done which I think I had discovered before.

It's a "force subscribe" with no confirmation needed, i.e., not a "double opt-in," and it is what the subscription pop-over uses, and has used successfully for months now.

All that to say, it works as it is intended to.

If you want to move the curl relay to its own domain, then I'm fine with that. I won't endorse putting services in a Bricolage-managed directory.

**Comment by phillipadsmith** [comment]
You'll see an explanation to your findings in the code linked above ;-)

https://github.com/TheTyee/webhook.thetyee.ca-subscribe/blob/master/app.pl#L46-L47

**Comment by MrBryan** [comment]
a-ha. Personally I would think the widget was broken if I didn't get a confirmation until the next edition and in theory someone/bot could subscribe anyone they like this way, but if there haven't been any problems I guess there is no reason we'd start having one.

Looks like it's setting both casl flags although they haven't seen/signed off on anything, is that OK too?

**Comment by phillipadsmith** [comment]
> a-ha. Personally I would think the widget was broken if I didn't get a confirmation until the next edition

Well, you can see how the interaction works here:
http://thetyee.ca/ui/js/subscription-offer.js

... basically, after submit is says "Thank you for subscribing! Now you're on the list. You can expect your Tyee email edition to arrive soon..."  

So the footer one should probably do the same if we go that route.

> and in theory someone/bot could subscribe anyone they like this way

Yes, indeed, that could happen... and yet... it has not happened. I think I did some poking around to see what other publishers were doing and found that the single opt-in was back in fashion. 

> Looks like it's setting both casl flags although they haven't seen/signed off on anything, is that OK too?

I walked through all of these subscription processes with Jeanette a few weeks back and I believe this is correct. If you want to triple check what's required on the form to meet the CASL requirements, that would be helpful. What we have on the pop-up ask is: "Your privacy is important to us. We will never share, rent, or sell your email address."

What do they see/sign-off on the other forms that is CASL specific?

**Comment by phillipadsmith** [comment]
Final note before I head out: I'm a-ok with the interaction you're proposing, just not with having random PHP services running in the Bricolage directory. If you just want to move it elsewhere, that's fine by me. I don't want to slow down the launch, but I do want to avoid making a mess of things if possible.

**Comment by MrBryan** [comment]
OK I'll leave it for now and use the line you used elsewhere about expecting it to come.

We show this on the subscribe.thetyee.ca page: 

> You'll also get early notice on Tyee events, news, promotions, partner messages and special initiatives. What's a partner message? A partner could be a carefully considered paid sponsor, or a group that we think you'd be interested in hearing from. We send these no more than once a month, and you are free to opt out of receiving partner messages. Simply having you on this list is a no-cost way to help fund more indie journalism.

Which I'm not sure is actually CASL compliant since it's kind of a "negative option" but the working around expressed consent is so vague in the legislation I'm not sure anyone could really say negative options are out.
